### DISCUSSION TITLE
Snappy Chrome 129 Causing Issues

### DISCUSSION THREAD

**Discussion by matthewhisle** [ujjwalguptaofficial/JsStore#375] [General]
Updated androids to chrome 129

Have been saving a JSSTORE database table of photo blobs..... New chrome release 129 this is now broken... SOmething to do with snappy compression, the insert doesnt throw an exception, but simply doesnt bulk insert my photos it is like it ignores it. Can we check on this?

**Comment by matthewhisle** [comment]
I got around this by doing one insert of one photo at a time in a loop, instead of bulk inserting them all at once. Not sure why this works and the other doesnt, but it does!

**Reply by ujjwalguptaofficial** [reply]
could you tell me the query ? I tried reproducing but could not. Please share your jsstore query.

**Reply by matthewhisle** [reply]
Yes thanks for looking into this.... 

So what you want to do, is download a bunch of images into blobs...

In the loop as you download each image, continually be adding them into a JS array... Something like var x = [{id:1234, img:blob},{id:1254, img:blob}]

Once the loop is done.... BULK INSERT the js var into a js store database table.... IF and only IF the images are too large, it will not insert anything, but will not throw an exception either... Then change your code to instead of in the loop building a JS array, have it be inserting each time 1 blob at a time into the database, and it'll work fine... Make sure the images are really large

**Reply by ujjwalguptaofficial** [reply]
interesting, i guess may be due to the blob size its stucked like huge memory being used and causing chrome to hang or something.

**Comment by ujjwalguptaofficial** [comment]
thanks for letting me know, i will look into this. Might be some API's are obsolete. 

Anyone facing these issue, please implement @matthewhisle  solution for now. I will provide the fix ASAP.
